packages/styled-components/src/models/StyledComponent.js

```diff
@@ -177,6 +179,21 @@ function useStyledComponentImpl<Config: {}, Instance>(
   return createElement(elementToBeCreated, propsForElement);
 }
 
+export function useStyle(styles: Styles, ...interpolations: Array<Interpolation>): string {
+  // We have to force interpolations to be functions because there's no way of knowing wether
+  // a prop is being interpolated. And if it is a prop but not a function, `generateAndInjectStyles`
+  // will fail to insert the new rule.
+  const rules = css(
+    isFunction(styles) ? styles : () => styles,
+    ...interpolations.map(i => (isFunction(i) ? i : (): Interpolation => i))
+  );
+  const id = useRef(generateId());
+  const componentStyle = new ComponentStyle(rules, id.current);
+  const theme = useContext(ThemeContext) || EMPTY_OBJECT;
+
+  return useInjectedStyle(componentStyle, false, { theme });
```

Comment:
Let's export `useInjectedStyle` and then isolate `useStyle` into its own file. Maybe it's time for us to have a `hooks` directory 🤔

Reply:
While working on this I expected there to be a `hooks` folder. If it's something y'all are open to then I would gladly make one and centralize hooks there. Thoughts?

Reply:
Definitely open to that, maybe a separate PR?